### ISSUE TITLE
Get rid of "flux" tuples and find good way to deal with flux based hardware constraints

### ISSUE THREAD

**Issue by AdriaanRol** [QE-Lab/OpenQL#176]
Flux tuples/edges were introduced in the CCLight config as a target of an operation. We have found that in experiment this is a very inconvenient (bad) way of specifying the target of a flux pulse. Furthermore, this is not a fundamental constraint of the underlying hardware (both CCLight and Transmon device). 

There are three main types of flux pulses (sorted by # target qubits) 
- Phase pulses/single qubit flux pulses 
- Two qubit flux pulses (such as the CZ) 
- Three qubit flux pules such as those performing a CZ on two qubits and moving another qubit out of the way. 

The first step is to get rid of the current (artificial) constraint, which requires any flux based operation to target an edge (and having two qubits as target). 
The second step would be to find a proper way of describing the intricacies of these constraints and how to best express this both in the configuration files and when writing a program. 

This subject requires a lot of discussion, and as such should be postponed until after #160 .

**Comment by imranashraf** [comment]
addressed in https://github.com/QE-Lab/OpenQL/issues/224. so closing.
